apispec/core.py

```diff
@@ -138,82 +133,81 @@ def schema(
             raise DuplicateComponentNameError(
                 'Another schema with name "{}" is already registered.'.format(name)
             )
-        ret = {}
+        component = component or {}
+        ret = component.copy()
         # Execute all helpers from plugins
         for plugin in self._plugins:
             try:
-                ret.update(plugin.schema_helper(name, definition=ret, **kwargs) or {})
+                ret.update(plugin.schema_helper(name, component, **kwargs) or {})
             except PluginMethodNotImplementedError:
                 continue
-        if properties:
-            ret["properties"] = properties
-        if enum:
-            ret["enum"] = enum
-        if description:
-            ret["description"] = description
-        if extra_fields:
```

Comment:
Ah, nice. This cleans things up quite a bit.

`special_topics.rst` will need to be updated--we can probably just get rid of the "Adding Additional Fields To Schema Objects" section altogether.